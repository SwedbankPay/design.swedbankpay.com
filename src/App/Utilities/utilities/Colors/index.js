import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const TextColors = () => (
    <>
        <h2 id="text-colors">Text colors</h2>
        <p>To use a color simply prepend <CodeTags type="secondary" code=".text-{name}" />.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p className="text-brand">.text-brand</p>
            <p className="text-default">.text-default</p>
            <p className="text-neutral">.text-neutral</p>
            <p className="text-success">.text-success</p>
            <p className="text-warning">.text-warning</p>
            <p className="text-danger">.text-danger</p>
            <p className="text-white bg-default">.text-white</p>
            <p className="text-black-50">.text-black-50</p>
            <p className="text-white-50 bg-default">.text-white-50</p>
        </ComponentPreview>

        <h3>With <CodeTags type="secondary" code=".text-muted" /></h3>
        <p>To get a muted effect on the color you are using, append a <CodeTags type="primary" code={"<span>"} />-element with the class <CodeTags type="secondary" code=".text-muted" /> to the desired element.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p className="text-brand">.text-brand <span className="text-muted">.text-muted</span></p>
            <p className="text-default">.text-default <span className="text-muted">.text-muted</span></p>
            <p className="text-neutral">.text-neutral <span className="text-muted">.text-muted</span></p>
            <p className="text-success">.text-success <span className="text-muted">.text-muted</span></p>
            <p className="text-warning">.text-warning <span className="text-muted">.text-muted</span></p>
            <p className="text-danger">.text-danger <span className="text-muted">.text-muted</span></p>
            <p className="text-white bg-default">.text-white <span className="text-muted">.text-muted</span></p>
        </ComponentPreview>
    </>
);

const BackgroundColors = () => (
    <>
        <h2 id="background-colors">Background colors</h2>
        <p>To alter the background-color of an element, add the class <CodeTags type="secondary" code="background-{name}" />.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="container">
                <div className="row">
                    <div className="col-12 text-white bg-brand p-2">.bg-brand</div>
                    <div className="col-12 text-white bg-default p-2">.bg-default</div>
                    <div className="col-12 text-white bg-neutral p-2">.bg-neutral</div>
                    <div className="col-12 text-white bg-success p-2">.bg-success</div>
                    <div className="col-12 text-white bg-warning p-2">.bg-warning</div>
                    <div className="col-12 text-white bg-danger p-2">.bg-danger</div>
                    <div className="col-12 bg-white p-2">.bg-white</div>
                </div>
            </div>
        </ComponentPreview>
    </>
);

const Colors = () => (
    <DocContainer docToc>
        <p className="lead">Convey meaning through color with a handful of color utility classes.</p>
        <TextColors />
        <BackgroundColors />
    </DocContainer>
);

export default Colors;

/* For testing */
export { TextColors, BackgroundColors };
