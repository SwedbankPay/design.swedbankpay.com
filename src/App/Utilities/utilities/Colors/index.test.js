import React from "react";
import { shallow } from "enzyme";

import Colors, { TextColors, BackgroundColors } from "./index";

describe("Utilities: Colors", () => {
    it("is defined", () => {
        expect(Colors).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Colors />);

        expect(wrapper).toMatchSnapshot();
    });

    it("TextColors renders", () => {
        const wrapper = shallow(<TextColors />);

        expect(wrapper).toMatchSnapshot();
    });

    it("BackgroundColors renders", () => {
        const wrapper = shallow(<BackgroundColors />);

        expect(wrapper).toMatchSnapshot();
    });
});
