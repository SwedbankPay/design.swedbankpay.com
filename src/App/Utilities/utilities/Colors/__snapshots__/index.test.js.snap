// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: Colors BackgroundColors renders 1`] = `
<Fragment>
  <h2
    id="background-colors"
  >
    Background colors
  </h2>
  <p>
    To alter the background-color of an element, add the class 
    <CodeTags
      code="background-{name}"
      type="secondary"
    />
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-white bg-brand p-2"
        >
          .bg-brand
        </div>
        <div
          className="col-12 text-white bg-default p-2"
        >
          .bg-default
        </div>
        <div
          className="col-12 text-white bg-neutral p-2"
        >
          .bg-neutral
        </div>
        <div
          className="col-12 text-white bg-success p-2"
        >
          .bg-success
        </div>
        <div
          className="col-12 text-white bg-warning p-2"
        >
          .bg-warning
        </div>
        <div
          className="col-12 text-white bg-danger p-2"
        >
          .bg-danger
        </div>
        <div
          className="col-12 bg-white p-2"
        >
          .bg-white
        </div>
      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Utilities: Colors TextColors renders 1`] = `
<Fragment>
  <h2
    id="text-colors"
  >
    Text colors
  </h2>
  <p>
    To use a color simply prepend 
    <CodeTags
      code=".text-{name}"
      type="secondary"
    />
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p
      className="text-brand"
    >
      .text-brand
    </p>
    <p
      className="text-default"
    >
      .text-default
    </p>
    <p
      className="text-neutral"
    >
      .text-neutral
    </p>
    <p
      className="text-success"
    >
      .text-success
    </p>
    <p
      className="text-warning"
    >
      .text-warning
    </p>
    <p
      className="text-danger"
    >
      .text-danger
    </p>
    <p
      className="text-white bg-default"
    >
      .text-white
    </p>
    <p
      className="text-black-50"
    >
      .text-black-50
    </p>
    <p
      className="text-white-50 bg-default"
    >
      .text-white-50
    </p>
  </ComponentPreview>
  <h3>
    With 
    <CodeTags
      code=".text-muted"
      type="secondary"
    />
  </h3>
  <p>
    To get a muted effect on the color you are using, append a 
    <CodeTags
      code="<span>"
      type="primary"
    />
    -element with the class 
    <CodeTags
      code=".text-muted"
      type="secondary"
    />
     to the desired element.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p
      className="text-brand"
    >
      .text-brand 
      <span
        className="text-muted"
      >
        .text-muted
      </span>
    </p>
    <p
      className="text-default"
    >
      .text-default 
      <span
        className="text-muted"
      >
        .text-muted
      </span>
    </p>
    <p
      className="text-neutral"
    >
      .text-neutral 
      <span
        className="text-muted"
      >
        .text-muted
      </span>
    </p>
    <p
      className="text-success"
    >
      .text-success 
      <span
        className="text-muted"
      >
        .text-muted
      </span>
    </p>
    <p
      className="text-warning"
    >
      .text-warning 
      <span
        className="text-muted"
      >
        .text-muted
      </span>
    </p>
    <p
      className="text-danger"
    >
      .text-danger 
      <span
        className="text-muted"
      >
        .text-muted
      </span>
    </p>
    <p
      className="text-white bg-default"
    >
      .text-white 
      <span
        className="text-muted"
      >
        .text-muted
      </span>
    </p>
  </ComponentPreview>
</Fragment>
`;

exports[`Utilities: Colors renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Convey meaning through color with a handful of color utility classes.
  </p>
  <TextColors />
  <BackgroundColors />
</DocContainer>
`;
